test/array.test.js

```diff
@@ -152,28 +153,56 @@ buildTest({
   '@data': ['test']
 })
 
-test('invalid items throw', (t) => {
+test('coerce number to string type item', (t) => {
```

Comment:
Is `patternProperties` still a thing?

Reply:
I don't know why there is a patternProperties in this test. It works without them ('invalid' string always doesn't match the object schema). The test will work differently after this PR, but not because of patternProperties.

How the test works now:
```js
test('invalid items throw', (t) => {
  t.plan(1)
  const schema = {
    type: 'object',
    properties: {
      args: {
        type: 'array',
        items: [
          {
            type: 'object',
            patternProperties: {
              '.*': {
                type: 'string'
              }
            }
          }
        ]
      }
    }
  }
  const stringify = build(schema)
  t.throws(() => stringify({ args: ['invalid'] })) // throws an error because 'invalid' is not an object
})
```

How it will work:
```js
test('invalid items throw', (t) => {
  t.plan(1)
  const schema = {
    type: 'object',
    properties: {
      args: {
        type: 'array',
        items: [
          {
            type: 'object',
            patternProperties: {
              '.*': {
                type: 'string'
              }
            }
          }
        ]
      }
    }
  }
  const stringify = build(schema)
  t.equal(stringify({ args: ['invalid'] }), '{"args":[{"0":"i","1":"n","2":"v","3":"a","4":"l","5":"i","6":"d"}]}') // it coerces 'invalid' string to the object schema
})
```

It will work just like the same schema with patternProperties works now without array.
```js
test('invalid items throw', (t) => {
  t.plan(1)
  const schema = {
    type: 'object',
    patternProperties: {
      '.*': {
        type: 'string'
      }
    }
  }
  const stringify = build(schema)
  t.equal(stringify('invalid'), '{"0":"i","1":"n","2":"v","3":"a","4":"l","5":"i","6":"d"}')
})

```

Reply:
@Eomm  I decided to remove the test instead of updating it because this coercion looks like nonsense. If you want I can add the test like that.

Reply:
I like the idea of removing all the "validation" logic here, but I would move it to the `next` branch.

I think the new output could warn a user that does not understand what is going on: I would prefer an empty output instead of `{"0":"i","1":"n","2":"v","3":"a","4":"l","5":"i","6":"d"}`, but we can't do it without keeping the "validation" logic tho

Reply:
The output you see isn't related to the array at all. It's all about how FJS coerces string to object.